geniza/corpus/annotation_export.py

```diff
@@ -0,0 +1,215 @@
+            # with all associated annotations, as a backup
+            # that could be imported into any w3c annotation server
+
+            for canvas, annotations in annos_by_canvas.items():
+                annolist_name = self.annotation_list_name(canvas)
+                annolist_out_path = os.path.join(output_dir, "%s.json" % annolist_name)
+                updated_filenames.append(annolist_out_path)
+                with open(annolist_out_path, "w") as outfile:
+                    json.dump(
+                        annotations_to_list(annotations, uri="test"), outfile, indent=2
+                    )
+
+            # load django template for rendering html export
+            html_template = get_template("corpus/transcription_export.html")
+
+            # for convenience and more readable versioning, also generate
+            # text and html transcription files
+            for edition in document.digital_editions():
+                # print(edition, edition.source)
+                # filename based on pgpid and source authors;
+                # explicitly label as transcription for context
+                base_filename = self.transcription_filename(document, edition.source)
+                for output_format in ["txt", "html"]:
+                    # put in the appropriate transription dir by format,
+                    # use format as file extension
+                    outfile_path = os.path.join(
+                        transcription_output_dir[output_format],
+                        "%s.%s" % (base_filename, output_format),
+                    )
+                    updated_filenames.append(outfile_path)
+                    with open(outfile_path, "w") as outfile:
+                        if output_format == "html":
+                            content = html_template.render(
+                                {"document": document, "edition": edition}
+                            )
+                            outfile.write(content)
+                        else:
+                            # text version is meant for corpus analytics,
+                            # so should be minimal and content only
+                            outfile.write(edition.content_text)
+
+        # prep updated files for commit to git repo
+        #  - adjust paths so they are relative to git root
+        updated_filenames = [
+            f.replace(base_output_dir, "").lstrip("/") for f in updated_filenames
+        ]
+        repo.index.add(updated_filenames)
+        if repo.is_dirty():
+            # print("Committing changes")
+            repo.index.commit("Automated data export from PGP")
+            if push_changes:
+                try:
+                    origin = repo.remote(name="origin")
+                    # pull any remote changes
+                    origin.pull()
+                    # push data updates
+                    # TODO: how to tell repo object not to output hashes?
```

Comment:
By output do you mean it logs to stdout?

Reply:
yes, it prints out some content — I think it's similar to what you'd get if you ran the git command yourself